### ISSUE TITLE
1.21 Save and Quit will crash

### ISSUE THREAD

**Issue by Spturate** [gnembon/fabric-carpet#1945]
When I join the server and then quit to join a single player game, and then quit the game, it will crash
1.21 Fabric 0.15.11
Mods list：
fabric API 0.100.4
carpet 1.4.147
log：[latest.log](https://github.com/user-attachments/files/16108093/latest.log)，
[错误报告-2024-7-5_17.48.41.zip](https://github.com/user-attachments/files/16108106/-2024-7-5_17.48.41.zip)

**Comment by acaciachan** [comment]
[crash-2024-07-06_09.34.11-client.txt](https://github.com/user-attachments/files/16114593/crash-2024-07-06_09.34.11-client.txt)
me, too.

**Comment by Ka1nnn** [comment]
yeah its been an issue for awhile now and idk why its still not fixed 🤷‍♂️ the fix is already in PR just not merged

https://github.com/gnembon/fabric-carpet/pull/1913

[fabric-carpet-1.21-1.4.147+v240623.zip](https://github.com/user-attachments/files/16114728/fabric-carpet-1.21-1.4.147%2Bv240623.zip)

you can compile the jar with the fix from altrisi's pr or use the one i provided which is just a compiled jar with altrisi's fix if you don't wanna bother with all that

**Comment by Spturate** [comment]
> 是的，它已经有一段时间了，为什么它仍然没有修复🤷 ♂️，修复程序已经在 PR 中，只是没有合并
> 
> #1913
> 
> [fabric-carpet-1.21-1.4.147+v240623.zip](https://github.com/user-attachments/files/16114728/fabric-carpet-1.21-1.4.147%2Bv240623.zip)
> 
> 您可以使用 Altrisi 的 PR 中的修复程序编译 jar，或者使用我提供的 jar，如果您不想打扰所有这些，它只是一个带有 Altrisi 修复程序的编译 jar

OK,thank you

**Comment by jonathangiha** [comment]
> yeah its been an issue for awhile now and idk why its still not fixed 🤷‍♂️ the fix is already in PR just not merged
> 
> #1913
> 
> [fabric-carpet-1.21-1.4.147+v240623.zip](https://github.com/user-attachments/files/16114728/fabric-carpet-1.21-1.4.147%2Bv240623.zip)
> 
> you can compile the jar with the fix from altrisi's pr or use the one i provided which is just a compiled jar with altrisi's fix if you don't wanna bother with all that

Thank you, this fixed the issue. I'll paste the error I was getting to make it easier for others to find this.

The game crashed whilst mouseclicked event handler Error: java.lang.NullPointerException: Cannot invoke "carpet.fakes.MinecraftServerInterface.getScriptServer()" because "server" is null

**Comment by gangber1** [comment]
me too, but1.20.6

[crash-2025-08-31_02.17.51-client.txt](https://github.com/user-attachments/files/22060623/crash-2025-08-31_02.17.51-client.txt)
